### ISSUE TITLE
`source_error` is just an empty type in the spec.... makes generations useless

### ISSUE THREAD

**Issue by jessfraz** [oxidecomputer/omicron#574]
<img width="426" alt="Screen Shot 2022-01-03 at 1 43 02 PM" src="https://user-images.githubusercontent.com/1445228/147983620-c93d728c-fc3e-466a-88d9-98decc8492b7.png">

**Comment by davepacheco** [comment]
Yeah, this certainly should be improved, one way or another.

I believe this type is used only in the API that fetches saga state.  This describes a case where an action failed.  The source error comes from the saga action, and it's allowed to be any JSON-serializable thing today.  That's why it has no more specific schema.

This is currently only a debugging API.  We may eventually want to put this information into the console in the future but we'll need to add some structure to do that.

I could see a few ways to go here:

* standardize on the specific error type that actions produce
* put these API endpoints on a separate dropshot server, or maybe just a different OpenAPI spec hosted on the same server?  This will hide them from people using the public API.
* hide these API endpoints from the spec (I don't like this option)

This is mainly a question of prioritization.

@jessfraz I get that this produces useless code for this type, but I expect that you're not planning to do anything with these debug endpoints and don't need this type.  Is that right?  Is it in the way for some other reason?

**Comment by jessfraz** [comment]
yeah i mean its fine, for go I made these an `interface{}` ie. anything and for rust I make these `serde json.Value` or whatever so like anything as well, its only annoying if its something customers touch but since it sounds like it isn't I wouldn't worry about it
